airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -509,6 +506,12 @@ def create_job_yaml(self, context):
                         'annotations': {
                             'cluster-autoscaler.kubernetes.io/safe-to-evict':
                             'false'
+                        },
+                        'labels': {
+                            'app.kubernetes.io/name': self.dag_id,
+                            'app.kubernetes.io/part-of': self.part_of,
```

Comment:
shouldn't this be:

```python
'app.kubernetes.io/part-of': self.part_of or self.dag.part_of or UNDEFINED_LABEL,
'k8s.bluecore.com/team': self.team or self.dag.team or UNDEFINED_LABEL,
'app.kubernetes.io/component': self.component or self.dag.component or UNDEFINED_LABEL,
```

Reply:
This logic is handled in the BaseOperator constructor

Reply:
This can't be handled in the constructor as the DAG is optional. You would need to handle this in a "setter" for DAG as well.